### PR TITLE
Feature: New Backend System Migration

### PR DESCRIPTION
**Pull request by ionSurf**
## Hey, I just made a Pull Request!

This feature provides full integration with the new backend system.

### Context

Given that Backstage is rapidly evolving, this plugin was in danger of falling behind and starting to have notable incompatibilities.

### Issue ticket number and link

- Fixes # (issue)

### Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have verified that the code builds perfectly fine on my local system
- [x] I have verified that my code follows the style already available in the repository
- [x] A changeset describing the change and affected packages. ([more info](https://github.com/AxisCommunications/backstage-plugins/blob/main/CONTRIBUTING.md#changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by anicke — CHANGES_REQUESTED** [review]
Thank you very much, I have some minor concerns!

File: .changeset/orange-buses-hug.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@axis-backstage/plugin-jira-dashboard-backend': major
+---
+
+Migrated Jira Dashboard Backend to new Backstage backend system
```
[the remark this answers is not in the document]
Comment (ionSurf) [reply .changeset/orange-buses-hug.md:5]: Yeah, happy to change it to that.

File: plugins/jira-dashboard-backend/src/service/router.ts [hunk 2]
```diff
@@ -1,98 +1,74 @@
...
+  type Filter,
+  type JiraResponse,
+  type Project,
+} from '@axis-backstage/plugin-jira-dashboard-common';
+import { getProjectAvatar } from '../api';
+import stream from 'stream';
 
 /**
  * Constructs a jira dashboard router.
- * @deprecated Please migrate to the new backend system as this will be removed in the future.
```
[the remark this answers is not in the document]
Comment (ionSurf) [reply plugins/jira-dashboard-backend/src/service/router.ts:38]: I'd say let's go ahead and remove the exports. You already had the deprecated tags in the latest version. According to the documentation, the next step is to remove the exports.

[the remark this answers is not in the document]
Comment (anicke) [reply plugins/jira-dashboard-backend/src/service/router.ts:38]: Agreed, I would be surprised if anyone was using the "old backend" still.

[the remark this answers is not in the document]
Comment (ionSurf) [reply plugins/jira-dashboard-backend/src/service/router.ts:38]: I'd even go as far as to stop functions that were previously used to create a legacy-wrapper to be exported by the plugin, as suggested by the docs . I'm commiting these changes too, let me know what you think.

[the remark this answers is not in the document]
Comment (ionSurf) [reply plugins/jira-dashboard-backend/src/service/router.ts:38]: I've noticed too that as a best practice, all types are usually exported from the common plugin. How about I move those to the types file in the common plugin and export all types from there?

[the remark this answers is not in the document]
Comment (anicke) [reply plugins/jira-dashboard-backend/src/service/router.ts:38]: Let's not move any types right now.

File: plugins/jira-dashboard-backend/src/index.ts [hunk 3]
```diff
@@ -4,9 +4,6 @@
  * @packageDocumentation
  */
 
-export * from './service/router';
 export { jiraDashboardPlugin as default } from './plugin';
-export { searchJira } from './api';
 export type { SearchOptions } from './api';
-export { jqlQueryBuilder } from './queries';
```
Comment (anicke) [inline plugins/jira-dashboard-backend/src/index.ts:11]: We can only remove createRouter and RouterOptions, i.e. what is deprecated. The others should be there.

Comment (ionSurf) [reply plugins/jira-dashboard-backend/src/index.ts:11]: OK. I'll put them back.

File: plugins/jira-dashboard-backend/src/api.ts [hunk 4]
```diff
@@ -93,8 +93,6 @@ export type SearchOptions = {
  * @param config - A Backstage config
  * @param jqlQuery - A string containing the jql query.
  * @param options - Query options that will be passed on to the POST request.
- *
- * @public
```
Comment (anicke) [inline plugins/jira-dashboard-backend/src/api.ts:97]: We need to add the "@public" here? The "searchJira" is not included in the "api-report"?

Comment (ionSurf) [reply plugins/jira-dashboard-backend/src/api.ts:97]: Right... I just aded it back.

**Comment by anicke** [comment]
You just need to commit the updates to "yarn.lock"!

**Review by anicke — APPROVED** [review]

**Comment by ionSurf** [comment]
Thank you @anicke for all the support and attention to this PR. Do you know when there'll be a new release that would include these changes?

**Comment by anicke** [comment]
It has been published to npm now! Thank you for the PR!

**Pull request merged**